exporter/sumologicexporter/filtering.go

```diff
@@ -0,0 +1,101 @@
+package sumologicexporter
+
+import (
+	"fmt"
+	"regexp"
+	"sort"
+	"strconv"
+	"strings"
+
+	"go.opentelemetry.io/collector/consumer/pdata"
+)
+
+type filtering struct {
+	regexes []*regexp.Regexp
+}
+
+// FieldsType represents concatenated metadata
+type FieldsType string
+
+func newFiltering(fields []string) (*filtering, error) {
```

Comment:
I'd change the pointer type on which we operate on to a value (accordingly in other invocations)
It's not a fat type and there are no changes being done on struct fields.

```suggestion
func newFiltering(fields []string) (filtering, error) {
```

Reply:
What to return in case of error? `Nil` is not a proper filtering structure. 🤔

Reply:
`filter{}`